lib/config.js

```diff
@@ -1,14 +1,43 @@
+'use strict';
+
 var config = {
-    brand: 'Discord',
-    token: process.env.OAUTH_TOKEN,
-    trackingID: process.env.TRACKING_ID,
-    commentWait: process.env.COMMENT_WAIT || 250,
-    commentAttempts: process.env.COMMENT_ATTEMPTS || 10,
-
-    // Used when no environment variable (process.env.PORT) is set
-    port: 3000,
-    host: 'localhost',
-    protocol: 'http:'
+    def: {
+        brand: 'Discord',
+        token: process.env.OAUTH_TOKEN,
+        commentAttempts: process.env.COMMENT_ATTEMPTS || 10,
+    },
+    development: {
+        port: 3000,
+        redisURL: 'redis://localhost',
+        databaseURL: 'postgres://localhost/discord_test'
+    },
+    test: {
+        protocol: 'http:',
+        host: 'localhost',
+        port: 3000,
+        trackingID: '654321',
+        commentWait: 0,
+        redisURL: 'redis://localhost',
+        databaseURL: 'postgres://localhost/discord_test',
+        database: 'discord_test'
+    },
+    production: {
+        port: process.env.PORT,
+        trackingID: process.env.TRACKING_ID,
+        commentWait: process.env.COMMENT_WAIT || 250,
+        redisURL: process.env.REDIS_URL,
+        databaseURL: process.env.DATABASE_URL
+    }
```

Comment:
Yeah, we can keep `lib/config.js` but we should remove the environment groupings to change it to something like:
```
var config = {
        brand: process.env.BRAND || 'Discord',
        rootUrl: process.env.ROOT_URL || 'http://localhost:3000',
        redisURL: process.env.REDIS_URL || 'redis://localhost',
        databaseURL: process.env.DATABASE_URL || 'postgres://localhost/discord_test',
        commentAttempts: process.env.COMMENT_ATTEMPTS || 10,
        commentWait: process.env.COMMENT_WAIT || 0,
        token: process.env.OAUTH_TOKEN,
        trackingID: process.env.GA_TRACKING_ID,
    }
};
```
(Note: I removed the `DATABASE` var because it can be generated from the `DATABASE_URL` var. Similarly, I combined PROTOCOL, HOST, and PORT vars into ROOT_URL.)

Once we do that, `.env` and/or `autoenv` are just convenience utilities that a developer can install if they want. (We could be nice and put `.env` in `.gitignore` for devs who **do** want to use it. :wink: )